### ISSUE TITLE
Do not treat as error when Client Token is not configured on initialize logger

### ISSUE THREAD

**Issue by thaim** [DataDog/browser-sdk#497]
When Client Token is not configured, datadog-logs treat as error and write
error message `Client Token is not configured, we will not send any data.`
to browser console on initializing.
It is usual situation that application start without Client Token
such as developing or testing environment.

It would be nice to write the message as warn in stead of error.

**Comment by bcaudan** [comment]
Hi @thaim,
 
What about not initialize browser logs if you not plan to use it?

**Comment by bcaudan** [comment]
Feel free to reopen if needed.

**Comment by cloudworkpro-dave** [comment]
Hey @bcaudan. Thanks for your work on this. I agree with @thaim on this I really don't think this should be throwing an error in the console. In our company, we have some people that have the environment variables set locally because they use the tool, and many other developers that don't, and even the developers that don't use the tool see this error warning. In either case, it's a configuration, people don't have it set on purpose so would this be an easy update to just change this console.error to console.warn. Would really appreciate it. Thank you!

**Comment by bcaudan** [comment]
Could you handle your use case by having a setup like:
```js
if (env.DATADOG_CLIENT_TOKEN) {
  DD_LOGS.init({ clientToken: env.DATADOG_CLIENT_TOKEN, ...})
}
```

or is there something more to it?

**Comment by alan-ps** [comment]
Hi @bcaudan,

The thing is validation happen (`validateAndBuildConfiguration`) before we decide which handler gonna be used ([createLogger](https://docs.datadoghq.com/logs/log_collection/javascript/#create-a-new-logger)). This means we should always specify `clientToken` (it's mandatory) to be able to use logger, even if we need `console` handler (e.g for dev environment), which does not require it.

The suggestion above does not work for such case, since we want to use logger (`console` handler), however - it's not possible w/o `clientToken`.

We can specify some random string (e.g 'empty') and everything works fine. However, I didn't find this in documentation so not sure it's correct.
```js
datadogLogs.init({
  clientToken: env.DATADOG_CLIENT_TOKEN || 'empty',
  ...
})

...
datadogLogs.createLogger('myLogger', {
  level: env.LOGGER_LEVEL,
  handler: env.NODE_ENV === 'prod' ? 'http' : 'console', 
})
```

In summary, it would be great to not specify `clientToken` for `console` handler if it's not used. Or maybe add a workaround with  'empty' string to the documentation. Does it make sense?
